frontend/src/abilities.ts

```diff
@@ -0,0 +1,74 @@
+import { AbilityBuilder } from "@casl/ability";
+import { store } from "@/store";
+
+export const subjects = {
+  TE: {
+    tasks: "TE/Tasks",
+    myTasks: "TE/MyTasks",
+    task: "TE/Task"
+  },
+  SQR: {
+    tasks: "SQR/Tasks",
+    form: "SQR/Form"
+  }
+};
+
+class Data {
+  constructor(props: any) {
+    Object.assign(this, props);
+  }
+}
+
+export class SQRForm extends Data {
+  static get modelName() {
+    return subjects.SQR.form;
+  }
+}
+
+export const SubjectsPlugin = {
+  install(Vue: any) {
+    Vue.prototype.$subjects = subjects;
+  }
+};
+
+const hasRole = (roles: any) => {
+  return store.getters["user/hasRole"](roles);
+};
+
+const handleTEPermissions = (can: any, cannot: any) => {
+  if (hasRole("TE.coordinator")) {
+    can("manage", subjects.TE.task);
+    can("manage", subjects.TE.tasks);
```

Comment:
Is it implied by the framework that `manage` includes all the other actions?

Reply:
Yes but now I that I think about it maybe `SQR.coordinator` should have manage actions instead of specific actions for SQR phase.